codeU1.py

```diff
@@ -1,51 +1,54 @@
-# -*- coding: utf-8 -*-
-"""
-Created on Fri May 19 15:10:45 2017
-
-@author: Luiza
-"""
 
 # -*- coding: utf-8 -*-
 """
 Created on Thu May 18 10:12:38 2017
-CodeU exercise 1
+CodeU exercise 1 refined
 @author: Luiza
+refined
 """
 
 # Task 1: Given 2 strings, decide if one of them is a permutation of the other
 # Return:
 #         1: if both strings are empty
 #         0: if one string is empty or they are not a permutation
-#         1: given trimmed strings form a permutation
+#         1: given trimmed  and lower-cased strings form a permutation
 
 import numpy as np
+import logging
+import unittest
    
 def is_permutation(str1,str2):
-    yes = "'" + str1 + "'" + " is a permutation of " + "'" + str2 + "'";
-    no = "'" + str1 + "'" + " is not a permutation of " + "'" + str2 + "'";
+    #yes = "'" + str1 + "'" + " is a permutation of " + "'" + str2 + "'";
+    #no = "'" + str1 + "'" + " is not a permutation of " + "'" + str2 + "'";
     str1 = str1.strip().lower()
     str2 = str2.strip().lower()
     if  not str1 and not str2:
-        print "Two empty strings are permutations"
+        #print "Two empty strings are permutations"
+        logging.warning("Two empty strings are permutations")
         return 1
     ord1 = [ord(a) for a in str1]
     ord2 = [ord(a) for a in str2]
     if np.prod(ord1)!=np.prod(ord2):

```

Comment:
I'm still a bit worried about this part. While this would be harder to hit with (certainly short) strings, it sounds plausible that the sum and prod checks would misidentify a non-permutation as a permutation:

```
>>> a = [6]
>>> b = [1, 2, 3]
>>> sum(a) == sum(b)
True
>>> prod(a) == prod(b)
True
```